src/lib/markbind/src/patches/htmlparser2.js

```diff
@@ -129,6 +148,189 @@ Tokenizer.prototype._stateText = function(c){
+
+		if (testChar === undefined) {
+			if (c === "/" || c === ">" || whitespace(c)) {
+				return tagIndex;
+			}
+		} else if (testChar === lowerCaseChar) {
+			matchingSpecialTags.push(tagIndex);
+		}
+	}
+
+	this._matchingSpecialTags = matchingSpecialTags;
+
+	return this._matchingSpecialTags.length > 0
+		? HAS_MATCHING
+		: NO_MATCH;
+};
+
+/**
+ * Changes the Tokenizer state to BEFORE_SPECIAL if the token matches one of
+ * the first characters of _specialTagNames.
+ */
+Tokenizer.prototype._stateBeforeTagName = function(c) {
+	if (c === "/") {
+		this._state = BEFORE_CLOSING_TAG_NAME;
+	} else if (c === "<") {
+		this._cbs.ontext(this._getSection());
+		this._sectionStart = this._index;
+	} else if (c === ">" || this._special !== SPECIAL_NONE || whitespace(c)) {
+		this._state = TEXT;
+	} else if (c === "!") {
+		this._state = BEFORE_DECLARATION;
+		this._sectionStart = this._index + 1;
+	} else if (c === "?") {
+		this._state = IN_PROCESSING_INSTRUCTION;
+		this._sectionStart = this._index + 1;
+	} else {
+		// We want special tag treatment to be xmlMode independent for MarkBind.
+		this._state = this.isFirstCharacterSpecialTagCharacter(c)
+			? BEFORE_SPECIAL
+			: IN_TAG_NAME;
+		this._sectionStart = this._index;
+	}
+};
+
+/**
+ * Changes the Tokenizer state to IN_TAG_NAME or BEFORE_SPECIAL state again depending
+ * on whether there are still matches in _matchingSpecialTags.
+ */
+Tokenizer.prototype._stateBeforeSpecial = function(c) {
+	const result = this.processSpecialFunctions(c);
+	if (result === HAS_MATCHING) {
+		this._nextSpecialTagMatchIndex += 1;
+		return;
+	}
+
+	// Reset for processSpecialClosingTagCharacter later
+	this._nextSpecialTagMatchIndex = 0;
+
+	this._state = IN_TAG_NAME;
+	this._index--; //consume the token again
+
+	if (result === NO_MATCH) {
+		return;
+	}
+
+	/*
+	 Assign the one-based index of the matching tag in the special tags array.
+	 This is due to htmlparser2 using 0 for SPECIAL_NONE,
+	 which would conflict with a zero-based index.
+	*/
+	this._special = result + 1;
+};
+
+/**
+ * Processes the _special flag and _nextSpecialTagMatchIndex state variable,
+ * returning a flag indicating whether the current special tag has finished matching or not.
+ */
+Tokenizer.prototype.processSpecialClosingTagCharacter = function(c) {
+	const nextTestChar = this.specialTagNames[this._special - 1][this._nextSpecialTagMatchIndex];
+
+	if (nextTestChar === undefined) {
+		this._nextSpecialTagMatchIndex = 0;
+		return (c === ">" || whitespace(c))
+			? HAS_MATCHED
+			: NO_MATCH;
+	} else if (nextTestChar === c.toLowerCase()) {
+		this._nextSpecialTagMatchIndex += 1;
+		return HAS_MATCHING;
+	}
+
+	this._nextSpecialTagMatchIndex = 0;
+	return NO_MATCH;
+};
+
+/**
+ * Changes the Tokenizer state to BEFORE_SPECIAL_END if the token matches one of
+ * the first character of the currently matched special tag.
+ */
+Tokenizer.prototype._stateBeforeCloseingTagName = function(c) {
+	if (whitespace(c));
```

Comment:
What do we do if it is whitespace?

Reply:
This line is from html parser2 as well; Its to allow something like `</    tagname>` to work